src/GlauberCollisionReader.cc

```diff
@@ -1,13 +1,11 @@
 #include "GlauberCollisionReader.hh"
 
-GlauberCollisionReader::GlauberCollisionReader() {
-}
-NucleonVector GlauberCollisionReader::GetNucleons() {
-    return nucleonVector;
+AAMCCinput GlauberCollisionReader::operator()() {
```

Comment:
Currently I just refactored GetNucleons() to operator()() since VCollisionReader has a pure virtual operator()() and GetNucleons()()'s signature matches the one needed. 
This commit wasn't focused on GlauberCollisionReader and thus doesn't substantially change it. In the end GlauberCollisionReader will contain a GMSTManager to retrieve nucleons_in directly, without the need to call Read() method externally. Just as any reader it will provide AAMCCinput when called, without the need to worry about its internal state.